### ISSUE TITLE
[GraphQL] Data Source

### ISSUE THREAD

**Issue by PascalSenn** [metabase/metabase#11333]
Hello there :wave: 

Metabase is a great way of providing data analytics to all employees of out company, which is great! 
Unfortunately most of our data is not in a database. We have an organisation wide graphql API that collects the data of many departments to provide a unified API.
It would just be incredible if we could use metabase to provide insights about the data in this graph. 

I've seen that @camsaul  mentioned that it is very unlikely that this will land in the core version of metabase. I just want to say that there is nothing like this for graphql. In the GraphQl world we have quite a bit of tooling for query metrics. Like execution time and things like this. But we do not have an easy way of giving insights into the data without doing json exports, custom power bi connectors or even custom react frontends.  GraphQL has a unified fullytyped interface, it doesn't matter which backend you use, queries look always the same. if there is one implementation provided every graphql user could use metabase! 
If you are interested to be part of the GraphQl ecosystem there is a massive potential there. Metabase could be a first class citizen of the GraphQL world. 
There is nothing like metabase, you would be the frist :) 

I was just wondering if there is any progress in this direction. 
Unfortunately I have no experience in clojure or any functional programming language. So I probably cannot really give any advise in how to integrate this into metabase. I do have quite a bit experience with graphql and may could help out if there are any questions.

I've seen a few issues around:
### GraphQL connector https://github.com/metabase/metabase/issues/6297

Closed by @kdoh
>I'm going to go ahead and close this issue to reflect the fact that we're unlikely to tackle a connector like this in the core metabase project. Feel free to keep discussion going since though since this is definitely an interesting area to explore and chat about further.

### GraphQL [PoC]  https://github.com/metabase/metabase/pull/7857 
Closed by @camsaul 
>Closing this and other old POC or WIP PRs for now to help clear the PR backlog. I'll leave the branch open in case we want to revisit this

### REST-API as datasource  https://github.com/metabase/metabase/issues/4831
Still Open

@tlrobinson  Opened a few very interesting issues for further investigations: 

### Standardize how we tell API endpoints to include sub-resources + remove functionality-specific custom endpoints https://github.com/metabase/metabase/issues/10035

Currently waiting on @camsaul  

and last but not least 
 ### "entities" optimizations and improvements https://github.com/metabase/metabase/issues/10044 

🥔

**Comment by camsaul** [comment]
#7857 was to use GraphQL for the Metabase API, i.e. for powering the frontend client, not for supporting GraphQL as a datasource

**Comment by PascalSenn** [comment]
@camsaul  sorry i totally missed that

**Comment by mazameli** [comment]
Hey, thanks for your enthusiasm for Metabase and GraphQL, @PascalSenn! We appreciate your thoughts. However, I do think this issue is a duplicate of #6297, and I'm going to close this as such, but we'd welcome your additional comments and thoughts on issue 6297. As @kdoh wrote there, it's unlikely something like this will be part of the OSS roadmap anytime soon given that we're currently focused on improving the stability, performance, and UX of the core parts of the product that are already in place.
